crawler/dockercontainer.py

```diff
@@ -72,13 +87,27 @@ def poll_docker_containers(timeout, user_list=None, host_namespace=''):
     containers.
     :return: a DockerContainer object (just the first container created).
     """
-
-    """
-    TODO: implement this function. For now, just emulate a failed poll:
-    no containers were created before timeout seconds.
-    """
     if timeout > 0:
-        time.sleep(timeout)
+        eventQ = multiprocessing.JoinableQueue()
+        eventProc = multiprocessing.Process(
+            name='docker-monitor',
+            target=start_polling, args=(eventQ, ))
```

Comment:
This will be called every iteration, and not once at the beginning of the program. Was that your intention? and in that case, wouldn't it be better to start this process at program init?